### PR TITLE
archive: add human-readable hint to Lchown error

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**- What I did**

Improved the text of Lchown error for rootless.

For #43576

**- How I did it**
See the code

**- How to verify it**

Before:
```
$ docker pull gcr.io/kubeflow-images-public/tensorflow-1.14.0-notebook-cpu:v0.7.0
failed to register layer: ApplyLayer exit status 1 stdout:  stderr: lchown /usr/local/bin/docker-credential-gcr: invalid argument
```

After:
```
$ docker pull gcr.io/kubeflow-images-public/tensorflow-1.14.0-notebook-cpu:v0.7.0
failed to register layer: ApplyLayer exit status 1 stdout:  stderr: failed to Lchown "/usr/local/bin/docker-credential-gcr" for UID 205001, GID 5000:
lchown /usr/local/bin/docker-credential-gcr: invalid argument (Hint: try increasing the number of subordinate IDs in /etc/subuid and /etc/subgid)
```

**- Description for the changelog**

archive: add human-readable hint to Lchown error

**- A picture of a cute animal (not mandatory but encouraged)**

🐧

### CODE REVIEW

File: pkg/archive/archive.go [hunk 1]
```diff
@@ -766,6 +768,10 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 			chownOpts = &idtools.Identity{UID: hdr.Uid, GID: hdr.Gid}
 		}
 		if err := os.Lchown(path, chownOpts.UID, chownOpts.GID); err != nil {
+			err = fmt.Errorf("failed to Lchown %q for UID %d, GID %d: %w", path, hdr.Uid, hdr.Gid, err)
+			if errors.Is(err, syscall.EINVAL) && userns.RunningInUserNS() {
+				err = fmt.Errorf("%w (Hint: try increasing the number of subordinate IDs in /etc/subuid and /etc/subgid)", err)
+			}
 			return err
```
[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply pkg/archive/archive.go:775]: Not sure.  https://github.com/pkg/errorshas been deprecated and archived

[the remark this answers is not in the document]
Comment (thaJeztah) [reply pkg/archive/archive.go:775]: Related to https://github.com/containerd/containerd/pull/6937#issuecomment-1125066122

basically; "native" error wrapping is not a real replacement for pkg/errors. I think @tonistiigi was considering to propose new maintainers (if any maintenance is needed for the package), and to request un-archiving it.

[the remark this answers is not in the document]
Comment (thaJeztah) [reply pkg/archive/archive.go:775]: @AkihiroSuda I had a quick chat with @tonistiigi on slack, and he also prefers updating to use `pkg/errors` (see his "thumbs up" above); could you make that change, then we should be ok to move forward with this.

[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply pkg/archive/archive.go:775]: done

File: pkg/archive/archive.go [hunk 2]
```diff
@@ -766,7 +768,11 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 			chownOpts = &idtools.Identity{UID: hdr.Uid, GID: hdr.Gid}
 		}
 		if err := os.Lchown(path, chownOpts.UID, chownOpts.GID); err != nil {
-			return err
+			msg := "failed to Lchown %q for UID %d, GID %d"
+			if errors.Is(err, syscall.EINVAL) && userns.RunningInUserNS() {
+				msg += " (try increasing the number of subordinate IDs in /etc/subuid and /etc/subgid)"
+			}
+			return errors.Wrapf(err, msg, path, hdr.Uid, hdr.Gid, err)
```
Comment (tonistiigi) [inline pkg/archive/archive.go:775]: one `err` too many

Comment (AkihiroSuda) [reply pkg/archive/archive.go:775]: fixed

**Review by tonistiigi — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**